### PR TITLE
Major documentation and config ovehaul

### PR DESCRIPTION
**Pull request by screamerbg**
With focus on guiding the user to add device management to their existing application, configure the device management feature and validate it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,72 @@
-# Simple Pelion Device Management Client
+# Mbed Device Management
 
 (aka Simple Pelion DM Client, SPDMC, previously called Simple Mbed Cloud Client)
```
Comment (MarceloSalazar) [inline README.md]: Maybe you can get rid of the aka Simple Pelion DM .....)?
IMO it shouldn't be shown, unless you point at the repos with all the history...

File: README.md [hunk 2]
```diff
@@ -1,31 +1,72 @@
...
+    $ mbed config -G CLOUD_SDK_API_KEY <your-api-key>
     ```
 
-2. Add your Pelion developer certificate to your project (`mbed_cloud_dev_credentials.c` file).
+3. Install the device management certificate:
 
-3. Reference the library from your main.cpp file, add network and storage drivers; finally initialize the Simple Pelion DM Client library. The is the architecture of a generic Simple Pelion DM Client application:
+    ```
+    $ mbed dm init -d "company.com" --model-name "product-model" -q --force
+    ```
```
Comment (MarceloSalazar) [inline README.md]: From the user (developer) point of view, I don't think we should be recommending `-q --force`

File: README.md [hunk 3]
```diff
@@ -1,31 +1,72 @@
...
+
+4. Reference the library from your `main.cpp` file, add network and storage drivers; finally initialize the SimpleMbedCloudClient library. The is the architecture of a generic device management application with Mbed OS:
```
Comment (MarceloSalazar) [inline README.md]: My preference would be to move step 4 (reference to library from main.cpp) to step 2, straight after importing the library; IMO it's the natural flow.

File: README.md [hunk 4]
```diff
@@ -56,15 +97,164 @@ This library is a simpler interface to Pelion DM Client, making it trivial to ex
     }
     ```
 
-## Example applications
+### Example applications
+
+  To help you start quickly, please refer to the following [generic application example](https://github.com/ARMmbed/pelion-ready-example). It demonstrates how to connect to the Pelion IoT Platform service, register resources and get ready to receive a firmware update.
+
+  Also, there are a number of board-specific applications that focus on providing more elaborate hardware features with Mbed OS and the Pelion IoT Platform. These are available in the Pelion [Quick-Start](https://cloud.mbed.com/quick-start).
+
```
Comment (MarceloSalazar) [inline README.md]: IMO it's very important that we add a list or table with information of Quick-start applications, so people can use it with Mbed CLI and other tools. This is the ideal place.
It should include: Platform(s) name, Connectivity, Storage, URL to example

File: README.md [hunk 5]
```diff
@@ -56,15 +97,164 @@ This library is a simpler interface to Pelion DM Client, making it trivial to ex
...
+```
+
+Refer to the next section about what tests are being executed.
+
+## Validation and Testing
+
+Mbed Device Management provides built-in tests to help you when define your device management configuration. Before running these tests, it's recommended that you run the `mbed dm init` command, which will install all needed credentials for both Connect and Update Pelion DM features. You can use the following command:
 ```
 $ mbed dm init -d "<your company name.com>" --model-name "<product model identifier>" -q --force
 ```
```
Comment (MarceloSalazar) [inline README.md]: This line is repeated 3 times in the doc. I think we should remove it and use the one in the 'Testing Setup' section below

File: README.md (hunk 4 above)
Comment (screamerbg) [reply README.md]: Do you have that list?

**Comment by screamerbg** [comment]
>     * Developers that want to get started with a Supported platform (not using the Quick-start, but just the SPDMC API).
> 
>     * Developers/Partners/Customers that want to enable a platform to be Pelion Ready (do the enablement, including testing)

I wouldn't split these two as configuration where you'd like to store your credentials, how to connect to a network etc, is part of the development flow. The difference is that's it's not made in CPP code but JSON.

File: README.md (hunk 4 above)
Comment (MarceloSalazar) [reply README.md]: See proposal here
https://gist.github.com/MarceloSalazar/9d2c5683277d598affa215dd8020d2e4

Comment (screamerbg) [reply README.md]: Now added

Comment (iriark01) [reply README.md]: If we add this to the docs, would you like these to be presented with the Import to IDE button?

File: README.md [hunk 6]
```diff
@@ -114,37 +336,47 @@ $ mbed dm init -d "<your company name.com>" --model-name "<product model identif
...
- 5. Run the Simple Pelion Device Management Client tests from the application directory:
+5. Run the Simple Pelion Device Management Client tests from the application directory:
 
-     ```mbed test -t <toolchain> -m <platform> --app-config mbed_app.json -n simple-mbed-cloud-client-tests-* --run -v```
+    ```
+    $ mbed test -t <toolchain> -m <platform> --app-config mbed_app.json -n simple-mbed-cloud-client-tests-* --run -v
+    ```
 
 ### Troubleshooting
 Below are a list of common issues and fixes for using Simple Pelion DM Client.
```
Comment (iriark01) [inline README.md]: Simple Pelion DM Client sounds like the correctly-branded name; is that what we should use throughout this document when we review?

File: README.md (hunk 4 above)
Comment (MarceloSalazar) [reply README.md]: I'd suggest we keep them without the import button. Having the url will let developers review the example's doc before deciding where to import (ide or cli)

**Comment by AnotherButler** [comment]
I didn't see this PR and created #71 instead. Could you please continue working there?

**Comment by screamerbg** [comment]
@AnotherButler Your PR was merged against this PR. Everything's good :)

**Comment by AnotherButler** [comment]
Thanks 👍

**Pull request merged**